lib/states/statesInRedis.js

```diff
@@ -208,111 +208,122 @@ class StateRedis {
+                    ready && typeof this.settings.disconnected === 'function' && this.settings.disconnected();
+                });
+
+                this.subSystem.on('error', error => {
+                    if (this.stop) {
+                        return;
                     }
-                    this.subSystem.on('end', () => {
-                        this.settings.connection.enhancedLogging && this.log.silly(this.namespace + ' States-Redis System Event end sub (stop=' + this.stop + ')');
-                        ready && typeof this.settings.disconnected === 'function' && this.settings.disconnected();
-                    });
+                    this.settings.connection.enhancedLogging && this.log.silly(this.namespace + ' Sub-Client States System No redis connection: ' + JSON.stringify(error));
+                });
 
-                    this.subSystem.on('error', error => {
-                        if (this.stop) {
-                            return;
-                        }
-                        this.settings.connection.enhancedLogging && this.log.silly(this.namespace + ' Sub-Client States System No redis connection: ' + JSON.stringify(error));
-                    });
+                if (this.settings.connection.enhancedLogging) {
+                    this.subSystem.on('connect', () =>
+                        this.log.silly(this.namespace + ' PubSub client States-Redis System Event connect (stop=' + this.stop + ')'));
 
-                    if (this.settings.connection.enhancedLogging) {
-                        this.subSystem.on('connect', () =>
-                            this.log.silly(this.namespace + ' PubSub client States-Redis System Event connect (stop=' + this.stop + ')'));
+                    this.subSystem.on('close', () =>
+                        this.log.silly(this.namespace + ' PubSub client States-Redis System Event close (stop=' + this.stop + ')'));
 
-                        this.subSystem.on('close', () =>
-                            this.log.silly(this.namespace + ' PubSub client States-Redis System Event close (stop=' + this.stop + ')'));
+                    this.subSystem.on('reconnecting', reconnectCounter =>
+                        this.log.silly(this.namespace + ' PubSub client States-Redis System Event reconnect (reconnectCounter=' + reconnectCounter + ', stop=' + this.stop + ')'));
+                }
 
-                        this.subSystem.on('reconnecting', reconnectCounter =>
-                            this.log.silly(this.namespace + ' PubSub client States-Redis System Event reconnect (reconnectCounter=' + reconnectCounter + ', stop=' + this.stop + ')'));
+                this.subSystem.on('ready', async _error => {
+                    try {
+                        this.subSystem && await this.subSystem.subscribe(`__keyevent@${this.settings.connection.options.db}__:expired`);
```

Comment:
Is there ever a case where `this.subSystem` is not defined inside the callback from its own events?

Reply:
Probably not and if so it’s caught anyway now..